src/plugins/oauth.js

```diff
@@ -0,0 +1,59 @@
+import qs from 'querystring';
+const TOKEN_REFRESHED = 'TOKEN_REFRESHED';
+
+export default {
+  onStart(request) {
+    if (this.refreshing) {
+      request.waitPromise = new Promise(resolve => {
+        this.client.eventEmitter.on(TOKEN_REFRESHED, () => {
```

Comment:
I was thinking that we might need to unsubscribe from this event as well in order to prevent from having this code called for every old request every time tokens are refreshed.